src/NServiceBus.Transport.SQS/Configure/SqsTransport.cs

```diff
@@ -226,12 +226,15 @@ public override async Task<TransportInfrastructure> Initialize(HostSettings host
             var topicCache = new TopicCache(SnsClient, hostSettings.CoreSettings, eventToTopicsMappings, eventToEventsMappings, topicNameGenerator, topicNamePrefix);
             var infra = new SqsTransportInfrastructure(this, hostSettings, receivers, SqsClient, SnsClient, QueueCache, topicCache, S3, Policies, QueueDelayTime, topicNamePrefix, EnableV1CompatibilityMode, DoNotWrapOutgoingMessages);
 
-            var queueCreator = new QueueCreator(SqsClient, QueueCache, S3, maxTimeToLive, QueueDelayTime);
+            if (hostSettings.SetupInfrastructure)
+            {
+                var queueCreator = new QueueCreator(SqsClient, QueueCache, S3, maxTimeToLive, QueueDelayTime);
 
-            var createQueueTasks = sendingAddresses.Select(x => queueCreator.CreateQueueIfNecessary(x, false))
-                .Concat(infra.Receivers.Values.Select(x => queueCreator.CreateQueueIfNecessary(x.ReceiveAddress, true))).ToArray();
+                var createQueueTasks = sendingAddresses.Select(x => queueCreator.CreateQueueIfNecessary(x, false))
+                    .Concat(infra.Receivers.Values.Select(x => queueCreator.CreateQueueIfNecessary(x.ReceiveAddress, true))).ToArray();
 
-            await Task.WhenAll(createQueueTasks).ConfigureAwait(false);
+                await Task.WhenAll(createQueueTasks).ConfigureAwait(false);
+            }
```

Comment:
Nitpicking, feel free to ignore. Is it better to pass the `setupInfrastructure` flag in the `QueueCreator` ctor? It simplifies this code, and the `QueueCrerator` already exposed "if"-based methods, like `CreateQueueIfNecessary`.

Reply:
I think it's clearer this way and in line with how the other transports do it. Also, it saves us having to create an instance of QueueCreator.